apps/one-drive/src/inngest/functions/data-protection/sync-items.ts

```diff
@@ -0,0 +1,302 @@
+import { logger } from '@elba-security/logger';
+import type { DataProtectionObject, DataProtectionObjectPermission } from '@elba-security/sdk';
+import { Elba } from '@elba-security/sdk';
+import { eq } from 'drizzle-orm';
+import { NonRetriableError } from 'inngest';
+import { env } from '@/env';
+import { inngest } from '@/inngest/client';
+import { db } from '@/database/client';
+import { organisationsTable } from '@/database/schema';
+import { decrypt } from '@/common/crypto';
+import type { MicrosoftDriveItem } from '../../../connectors/share-point/items';
+import { getItems } from '../../../connectors/share-point/items';
+import type { MicrosoftDriveItemPermissions } from '../../../connectors/share-point/permissions';
+import { getItemPermissions } from '../../../connectors/share-point/permissions';
+
+export type ItemsWithPermisions = {
+  item: MicrosoftDriveItem;
+  permissions: MicrosoftDriveItemPermissions[];
+};
+
+export const parseItems = (
+  inputItems: MicrosoftDriveItem[]
+): {
+  folders: MicrosoftDriveItem[];
+  items: MicrosoftDriveItem[];
+} => {
+  const folders: MicrosoftDriveItem[] = [];
+  const items: MicrosoftDriveItem[] = [];
+
+  inputItems.forEach((item) => {
+    if (item.folder) {
+      folders.push(item);
+    } else {
+      items.push(item);
+    }
+  });
+
+  return {
+    folders,
+    items,
+  };
+};

```

Comment:
Having items that could items or folders is a bit confusing. Are items items file ? If it's the case let's call them files.

```suggestion
export const groupItems = (
  items: MicrosoftDriveItem[]
) => items.reduce((acc, item) => {
  if (item.file) {
    acc.files.push(item)
  } else {
    acc.folders.push(item)
  }
  return acc
}, { files: [], folders[] });
```

Reply:
Items are not exactly files, they can be files and folders, also in Microsoft it called items, but we can call it files, like objects to get permissions and send to elba. Good suggestion with reduce.
Updated this